tests/unit_tests/test_controller_utils.py

```diff
@@ -37,10 +37,6 @@ def test_get_controller_resources_spot(controller_type,
                                        custom_controller_resources_config,
```

Comment:
Can we also add some unit tests for the automatically choosing cloud for controllers when the task resources are set.

Reply:
Added. Thanks! PTAL 🫡